src/components/ui/ToggleButton.jsx

```diff
@@ -13,15 +13,20 @@ const toggleSize = {
   },
 };
 
-const ToggleButton = ({ size = "m", checked, onToggle }) => {
+const ToggleButton = ({ size = "m", checked, onToggle, disabled = false }) => {
   const { buttonSize, thumbSize, activeTranslate } = toggleSize[size];
 
   return (
     <Button
+      disabled={disabled}
       className={`relative cursor-pointer rounded-full transition-colors duration-200 ${
-        checked ? "bg-black" : "bg-neutral-300"
+        disabled
+          ? "cursor-not-allowed bg-neutral-200"
+          : checked
+            ? "bg-black"
+            : "bg-neutral-300"
       } ${buttonSize}`}
-      onClick={onToggle}
+      onClick={disabled ? undefined : onToggle}
```

Comment:
```suggestion
      onClick={onToggle}
```
버튼이 disabled 상태일 때 onClick이 실행되지 않게 명시적으로 코드를 잘 작성해주셨네요!

다만, `disabled` 속성 자체가 클릭 이벤트를 막아주는 것이 기본 동작이라
`onClick`에서 별도로 `undefined`를 처리하지 않아도 정상적으로 작동하기 때문에
가독성과 유지보수 측면에서 `onClick={onToggle}` 형태로 간단하게 작성하는 것도 고려해볼 수 있을 것 같아요.

두 분은 어떤 방식이 더 좋다고 생각하시나요?
@Eun0713 @devjaeyoon

Reply:
좋은 리뷰 감사드립니다! 저는 희주님의 제안이 더 좋은 것 같네요!

Reply:
저도 개인적으로는 간결한 `onClick={onToggle}` 쪽이 더 좋은 거 같아요 :)

Reply:
그러면 희주님의 제안으로 수정하겠습니다! 두 분 모두 감사드립니다!